docker_entrypoint.sh

```diff
@@ -40,189 +62,304 @@ ADV_TOR_MAXSOCKSCONNS=$(yq e '.advanced.tor.maxsocksconns' ${BITMONERO_DIR}/star
-sed -i "s/MONERO_ANON_INBOUND_HOST/$MONERO_ANON_INBOUND_HOST/g" $new_conf_template
-sed -i "s/ADV_TOR_MAXONIONCONNS/$ADV_TOR_MAXONIONCONNS/g" $new_conf_template
+sed -i "s|MONEROD_BIND_IP|$MONEROD_BIND_IP|g" $config_file
+sed -i "s|PEER_TOR_ADDRESS|$PEER_TOR_ADDRESS|g" $config_file
+sed -i "s|TOR_PORT|$TOR_PORT|g" $config_file
+sed -i "s|ADV_TOR_MAXSOCKSCONNS|$ADV_TOR_MAXSOCKSCONNS|g" $config_file
+sed -i "s|RPC_TOR_ADDRESS_RESTRICTED|$RPC_TOR_ADDRESS_RESTRICTED|g" $config_file
+sed -i "s|RPC_TOR_ADDRESS|$RPC_TOR_ADDRESS|g" $config_file
+sed -i "s|MONERO_P2P_PORT_LOCAL_BIND|$MONERO_P2P_PORT_LOCAL_BIND|g" $config_file
+sed -i "s|MONERO_P2P_PORT|$MONERO_P2P_PORT|g" $config_file
+sed -i "s|MONERO_RPC_PORT_RESTRICTED|$MONERO_RPC_PORT_RESTRICTED|g" $config_file
+sed -i "s|MONERO_RPC_PORT|$MONERO_RPC_PORT|g" $config_file
+sed -i "s|MONEROD_LOCAL_HOST|$MONEROD_LOCAL_HOST|g" $config_file
+sed -i "s|ADV_TOR_MAXONIONCONNS|$ADV_TOR_MAXONIONCONNS|g" $config_file
+sed -i "s|TOR_SOCKS_PROXY_HOST|$TOR_SOCKS_PROXY_HOST|g" $config_file
 
 #PRUNING config:
 if [ "$ADV_PRUNING_MODE" = "true" ] ; then
- echo -e "\n# PRUNING\nprune-blockchain=1" >> $new_conf_template
+ echo -e "\n# PRUNING\nprune-blockchain=1" >> $config_file
  #if [ "$ADV_PRUNING_SYNCPRUNEDBLOCKS" = "true" ] ; then
- # echo "sync-pruned-blocks=1" >> $new_conf_template
+ # echo "sync-pruned-blocks=1" >> $config_file
  #fi
 fi
  
 #CUSTOM NODES config:
-echo -e "\n# CUSTOM NODES" >> $new_conf_template
-i=1
-num_custom_peers=$(yq e '.advanced.p2p.peer[]|length' ${BITMONERO_DIR}/start9/config.yaml | wc -l)
-while [[ $i -le $num_custom_peers ]] ; do
- peer_hostname=$(yq e '.advanced.p2p.peer[$i].hostname' ${BITMONERO_DIR}/start9/config.yaml)
- peer_port=$(yq e '.advanced.p2p.peer[$i].port' ${BITMONERO_DIR}/start9/config.yaml)
- peer_priority=$(yq e '.advanced.p2p.peer[$i].prioritynode' ${BITMONERO_DIR}/start9/config.yaml)
+echo -e "\n# CUSTOM NODES" >> $config_file
+i=0
+num_custom_peers=$(yq e ".advanced.p2p.peer[]|length" ${BITMONERO_DIR}/start9/config.yaml | wc -l)
+while [[ $i -le $(expr $num_custom_peers - 1) ]] ; do
+ peer_hostname=$(yq e ".advanced.p2p.peer[$i].hostname" ${BITMONERO_DIR}/start9/config.yaml)
+ peer_port=$(yq e ".advanced.p2p.peer[$i].port" ${BITMONERO_DIR}/start9/config.yaml)
+ peer_priority=$(yq e ".advanced.p2p.peer[$i].prioritynode" ${BITMONERO_DIR}/start9/config.yaml)
  if [ "$ADV_P2P_STRICTNODES" = "true" ] ; then
-  echo "add-exclusive-node=$peer_hostname:$peer_port" >> $new_conf_template
+  echo "add-exclusive-node=$peer_hostname:$peer_port" >> $config_file
  else
   if [ "$peer_priority" = "true" ] ; then
-   echo "add-priority-node=$peer_hostname:$peer_port" >> $new_conf_template
+   echo "add-priority-node=$peer_hostname:$peer_port" >> $config_file
   else
-   echo "add-peer=$peer_hostname:$peer_port" >> $new_conf_template
+   echo "add-peer=$peer_hostname:$peer_port" >> $config_file
   fi
-  echo "" >> $new_conf_template
+  echo "" >> $config_file
  fi
```

Comment:
This looks like it will cause problems

Reply:
Is there a method to reverse this?

Reply:
Yes... the config template and final config are built every time monero starts based on the config the user chooses in the web-UI.  If they make their peers exclusive or priority nodes, this will reflect their preference in the config.  If they un-check it in the config, it goes back to normal...